internal/repository/repository.go

```diff
@@ -471,6 +471,9 @@ func (r *Repository) LoadIndex(ctx context.Context) error {
 		return err
 	}
 
+	// remove obsolete indexes
+	validIndex.Sub(r.idx.Obsolete())
```

Comment:
Wouldn't that cause the obsolete indexes to be downloaded over and over again? After all these are still stored in the repository.